data-structures/binary_search_tree/bst.py

```diff
@@ -0,0 +1,109 @@
+class Node:
+    def __init__(self, value):
+        self.val = value
+        self.left = None
+        self.right = None
+
+
+class BST:
+    """
+    Binart Search Tree has 3 methods to sort nodes byself.
+    .inorder(), .preorder(), and .postorder()
+    """
+    def __init__(self, arr=()):
+        self.root = None
+        if len(arr):
+            for element in arr:
+                self.insert(element)
+
+    def in_order(self, operation):
+        """
+        In order traversal. Traverses nodes starting from head down to the left
+        node until val is None than continues to move down right and repeats.
+        """
+        def _walk(node=None):
+            """walk finction jump from node to node"""
+            if node is None:
+                return
+
+            if node.left is not None:
+                _walk(node.left)
+            operation(node.val)
+
+            if node.right is not None:
+                _walk(node.right)
+
+        if self.root is None:
+            return False
```

Comment:
You should not be returning `False` if the tree is empty. Since you are handling an empty tree in your `_walk()` function you actually do not even need this condition.